subsys/bluetooth/services/bas_c.c

```diff
@@ -272,3 +377,21 @@ int bt_gatt_bas_c_get(struct bt_gatt_bas_c *bas_c)
 	}
 	return bas_c->battery_level;
 }
+
+int bt_gatt_bas_c_read_interval_set(struct bt_gatt_bas_c *bas_c, s32_t interval,
+				    bt_gatt_bas_c_read_cb func)
+{
+	if (!bas_c || !func) {
+		return -EINVAL;
+	}
+
+	if (bt_gatt_bas_c_notify_supported(bas_c)) {
+		return -EBUSY;
+	}
+
+	bas_c->read_val.read_cb = func;
+	atomic_set(&bas_c->read_val.interval, interval);
```

Comment:
I don't think there is any point in having atomic operations on the interval. Either the periodic_read_process function take the new value or the old value.

Reply:
This variable is accessed from two different context so I am not sure that this is not needed.

Reply:
So is `bas_c->read_val.read_cb`, it depends on how you use it.